src/event-notion/index.ts

```diff
@@ -324,91 +325,138 @@ export const pingForTAPandCSIDeadlines = async (
+          'message': '\n⚠️ __**Booking confirmation for AS Funding is due in 1 week!**__ ⚠️\n',
+          'prop': 'Booking',
+          'propStatus': ['Booking TODO', 'Booking In Progress'],
+        }
+      }
+    },
+    {
+      'title': 'TAP Invoice Deadline',
+      'colour': 'ORANGE',
+      'dates': {
+        21: {
+          'pingRoles': ['fundingTeamID'],
+          'date': dateTimeNow.plus({ days: 21 }),
+          'message': '\n __**Invoice for TAP is due in a week!**___ \n',
+          'prop': 'TAP',
+          'propStatus': ['TAP TODO'],
+        },
+        16: {
+          'pingRoles': ['fundingTeamID'],
+          'date': dateTimeNow.plus({ days: 16}),
+          'message': '\n __**Invoice for TAP is due in 3 days!**__ \n',
+          'prop': 'TAP',
+          'propStatus': ['TAP In Progress', 'TAP TODO'],
+        },
+        14: {
+          'pingRoles': ['fundingTeamID', 'logisticsTeamID'],
+          'date': dateTimeNow.plus({ days: 14}),
+          'message': '\n⚠️ __**Invoice for TAP is due today!**__ ⚠️\n',
+          'prop': 'TAP',
+          'propStatus': ['TAP In Progress'],
+        }
+      }
+    },
+    {
+      'title': 'Event Details Confirmation',
+      'colour': 'YELLOW',
+      'dates': {
+        21: {
+          'pingRoles': ['logisticsTeamID', 'marketingTeamID'],
+          'date': dateTimeNow.plus({ days: 21 }),
+          'message': '\n __**Double-check venue, time, food, title and description for the event!**__\n',
+          'prop': 'PR',
+          'propStatus': ['PR TODO', 'PR In Progress'],
+        },
+        14: {
+          'pingRoles': ['logisticsTeamID'],
+          'date': dateTimeNow.plus({ days: 14}),
+          'message': '\n⚠️ __**Final check for all event details!**__ ⚠️\n',
+          'prop': 'PR',
+          'propStatus': ['PR TODO', 'PR In Progress'],
+        }
+      }
+    },
+  ];
+
+  // Used to compile all queries by iterating over above array  
+  var daysTrack = new Array<any>();
+  var dayQuery = new Array<any>();
+  var statusQuery = new Array<any>();
+  var props = new Array<string>();
+  var propStatuses = new Array<string>();
+
+  // Creates filter for all days equal to one of the above deadlines
+  // Slightly jankily adds 1 day to the original deadline, just to be sure to get 24 hours buffer if there's a weird start time
+  for(var i=0; i < daysToPing.length; i++){
+    for(var amountDays in daysToPing[i]['dates']){
+      dayQuery.push({
+        property: 'Date', 
+        date: {
+          equals: daysToPing[i]['dates'][amountDays]['date'].plus({days:1}).toISODate()
+        }}
+      );
+    }
+  }
+
+  // Creates filter for all statuses equal to one of the above deadlines
+  // Additionally, compile string for status selectors for isEventPingable to use
+  for(var i=0; i < daysToPing.length; i++){
+    for(var day in daysToPing[i]['dates']){
+      if(daysTrack.indexOf(day) == -1){
+        var cur = daysToPing[i]['dates'][day];
```

Comment:
We can probably use `.map` here to avoid using `for` loops since they're less efficient.

ex.
```ts
daysToPing.map((day) => day.dates.map(amountDays => ...));
```